arviz/utils.py

```diff
@@ -105,7 +112,9 @@ def _subset_list(subset, whole_list, filter_items=None, warn=True):
             )
 
         excluded_items = [
-            item[1:] for item in subset if item.startswith("~") and item not in whole_list
+            item[1:]
+            for item in subset
+            if _repr_str(item).startswith("~") and item not in whole_list
```

Comment:
Do we want the `~` effect to work on non-string variables? Can it really be used in those cases?

Reply:
I think the subset feature probably shouldn't work on non-string variables but I think the idea of extending similar functionality to them.